### DISCUSSION TITLE
Suavizado de vegetación

### DISCUSSION THREAD

**Discussion by agustincaballero** [manudona92/MapantES#24] [General]
He hecho una prueba para suavizar la vegetación, es decir, intentar que no tenga tantos ángulos rectos sino que sea más "curvada",  y que así tenga un aspecto algo más natural.
Esto es el antes:
![pullautus_veg mapant SIN_SUAVIZAR](https://user-images.githubusercontent.com/18677366/103762421-8d993f80-5018-11eb-9477-b291d41ceb74.png)

Y esto el después:
![pullautus_veg mapant](https://user-images.githubusercontent.com/18677366/103762455-9b4ec500-5018-11eb-9293-c95d010d08c4.png)

Echadle un vistazo a ver qué os parece, yo creo que queda bastante bien.
Lo único es que rebaja un poco la intensidad de los verdes más oscuros.

**Comment by manudona92** [comment]
Muy chulo Agustín!

De lo que he mirado:

- Los verdes más saturados no los rebaja tanto, quizá el más suave un poco. De todos modos, los pullauta.ini dejaré lightgreentone=200, que es para que el verde más claro sea un poco más saturado de salida. 

- Veo que no convierte los píxeles aislados, por ejemplo, un sólo (1) pixel blanco rodeado de amarillo, lo borra. Le he estado echando un vistazo a los cultivos para ver si los árboles eran "mono" píxel, pero he visto que en zonas de almendros y olivos, se hacen clusters de 3-4 pixels lo que los suavizará como un círculo. En resumen, que borre esos píxeles aislados no afecta.

A mí me gusta, le da un toque más realista. Por curiosidad ¿Es una herramienta raster del GDAL?

**Reply by agustincaballero** [reply]
Lo hago con ImageMagick, una librería para manipulación de imágenes

**Comment by jarufe** [comment]
A mi me gusta y me parece que da un resultado más adecuado a lo que queremos ofrecer.

**Comment by agustincaballero** [comment]
La duda que me surge ahora es que la vegetación tiene un montón de tonos de verde (vienen ya así del KP; no son productos del suavizado); ¿sería mejor ceñirse a los tres tonos de verde que marca la ISOM? 
Entiendo que los que habéis batallado más con el KP ya os lo habréis planteado, pero a mí me ha surgido ahora la duda.

**Comment by manudona92** [comment]
Sí, estoy al tanto. 

Aquí hay varias opciones: 

- Cambiarlo desde KP (lo más factible). En PULLAUTA.INI estoy cambiando los niveles de sombreados de verde a greenshades=0.8|1.8|3|99|99|99, el resto de parámetros los he dejado intactos. El resultado, con toda sinceridad, no es muy bueno. Lo hice con un ejemplo de Galicia ya que tiene muchos verdes. Lo que hace es rebajarlo, y dejar más blanco 

![image](https://user-images.githubusercontent.com/76484741/104061096-3d78d380-51f8-11eb-9d8b-fc2836a2de53.png)

- Modificarlo desde servidor, con una herramienta GIS o de edición de imagen y que haga una reclasificación. Yo lo hice con ARCGIS hace ya 5 años, reclasificando los 10-12 sombreados de  verdes a 3 clases ISOM. Estoy es más trabajo para servidor, y para Agustín. Lo dejaría como último plan 

- Dejarlo como esta, y viendo si con el suavizado que se esta haciendo se disimula más los niveles de verde. 

Aún asi, estoy escribiendo a Jarkko, a ver si me da una pista de como lo hicieron en mapant Finlandia (solo tienen 3 verdes) y/o que parámetros de KP puedo cambiar, sin modificar la estructura principal de los pullauta.ini que ya tenemos.

**Comment by agustincaballero** [comment]
Entiendo que no debe ser gran problema reducir esos colores al hacer el renderizado. de manera que conservemos el vegetation.png con toda la gama por si fuera útil en un futuro.
En cualquier caso, me parece bien preguntarles a los finlandeses.

**Reply by manudona92** [reply]
He podido dejarlo en 3 tonalidades de verde, lo único es que todas 3 tienen saturaciones parecidas y no se distinguen mucho entre ellas. Sigo esperando respuesta de Jarkko respecto como poder representar el verde 3 (100% impasable) con la configuracion actual, si es que realmente nos interesara poner el verde 3 en el mapant, que creo que dijimos que si

Izquierda es con 3 tonalidades y derecha es el antiguo con muchas tonalidades

![image](https://user-images.githubusercontent.com/76484741/104120459-5d62e100-5337-11eb-8a37-5a118add2447.png)

**Comment by agustincaballero** [comment]
Esto de los colores es que tiene más miga de lo que parece, jeje.
¿Puedes poner aquí la imagen de la izquierda original, sin unirla a la otra? Es que al unirla con la otra la paleta cambia. 
A mí me gusta más lo de la izquierda (pocos verdes) que lo de la derecha, ese difuminado no es muy de orientación.
De todas formas, los colores de la ISOM son éstos (que parece que tienen más contraste que la imagen de la izquierda)
![image](https://user-images.githubusercontent.com/18677366/104121632-10373d00-5340-11eb-8ee5-3ae5f0c3d5b1.png)

Una opción que me parece viable (y conforme la pienso, más adecuada me parece) es dejar que KP genere los múltiples tonos (lo de la derecha, esa imagen la guardaríamos), y al renderizar, que el cliente reduzca los colores hasta los tres ISOM. De esa manera podríamos hacer diferentes pruebas para ver cómo queda mejor, pudiendo volver a renderizar cuando queramos.

**Reply by manudona92** [reply]
Yo creo que el contraste de verde no es el mejor en la última prueba. Espero que Jarkko nos pueda dar un truco para ampliar el contraste de los verdes (que sean más ISOM) desde KP. Ahi va un ejemplo de solo los 3 verdes que he conseguido hasta ahora. 

![image](https://user-images.githubusercontent.com/76484741/104123188-dff49c00-5349-11eb-8584-733510bd37d1.png)

Respecto "que el cliente reduzca los colores hasta los tres ISOM", ¿A que te refieres? Entiendo que a través de una herramienta de imagen que genere los clusters?

**Reply by agustincaballero** [reply]
Me refiero a que en el cliente, antes de renderizar, se coja el png de vegetación y se cambien cada color verde por el equivalente de la ISOM que sea más parecido. Creo que es factible, aunque como sería por color "parecido", no sé si el resultado sería muy preciso.

**Reply by manudona** [reply]
Os pongo por aqui dos pequeños ejemplos de las últimas pruebas de los verdes: 

El de la derecha, es el que preferiria. He subido un poquito el verde 1 y bajado el verde 2, que en el de la izquierda veia el verde 2 más oscuro y no tan equilibrados. Es mi opinión pero podéis juzgas vosotros 

![image](https://user-images.githubusercontent.com/17010090/104323017-efa8e780-54e5-11eb-9bd8-532f6f7c0ba6.png)

Y el de la derecha comparado con "no los mejores ejemplos" de verde ISOM

![image](https://user-images.githubusercontent.com/17010090/104323191-2da60b80-54e6-11eb-8c21-06a4fef094d0.png)

y de la izquierda.

![image](https://user-images.githubusercontent.com/17010090/104323282-48788000-54e6-11eb-8eba-2b5a160eca16.png)


![pullautus_depr_0 150 8991 8](https://user-images.githubusercontent.com/17010090/104322584-6abdce00-54e5-11eb-85e6-905f94fefed0.png)
![pullautus_depr_0 150 81](https://user-images.githubusercontent.com/17010090/104322593-6c879180-54e5-11eb-9ba1-19cc5a480864.png)

Cuando lo decidamos, podemos ver como quedaria con el suavizado que  hace Agustín, y aplicarlo a los próximos test de El Hierro, Oia y Coto Maravillas. ( os tendria que actualizar el INI)

Se pueden cambiar un poco más desde KP los verdes, no habria problema
